website2/documentation/concepts.mdx

```diff
@@ -0,0 +1,153 @@
+bottom in the [Projects](#projects) section.
+
+All of these concepts are implemented as typescript types. Feel free to browse
+the [API docs](https://doc.deno.land/https://garn.io/ts/v0.0.19/mod.ts) as well.
+
+## Environments
+
+(See also the [API docs](https://doc.deno.land/https://garn.io/ts/v0.0.19/mod.ts/~/Environment).)
+
+`Environment`s define what is available to a computation. For example they can
+contain compilers and developer tools that you want to use on a project, but
+they may also define other things like environment variables that need to be
+set.
+
+Running `garn enter` in a shell will add whatever the `Enviroment` provides to
+the shell environment that you're in. That means you can still use your local
+developer tools, like your editor, etc., but will also have access to everything
+provided by the `Environment` you entered.
+
+The `garn` library provides a special environment called `emptyEnvironment`. As
+you might have guessed it's an `Environment` that doesn't provide anything.
+
+`Environment`s can also be used to create other things that then make use of
+what the `Environment`s provide. For example `Check`s, which are our next topic.
+
+## Checks
+
+(See also the [API docs](https://doc.deno.land/https://garn.io/ts/v0.0.19/mod.ts/~/Check).)
+
+`Check`s are commands (usually shell commands) that are used to define
+reproducible, automated tests or other checks. They run in a sandbox, which
+means they don't have access to your locally installed tools, nor to your local
+file system. Instead they are always based on an underlying `Environment`, and
+through that will have access to exactly the tools that the environment
+provides. In addition, a _copy_ of your source files is provided in the sandbox.
+That gives you the ability to define automated `Check`s on your source files.
+
+You can run `Check`s with `garn check`.
+
+Running `Check`s in a sandbox has its downsides, for example some tests require
+internet access. The big advantage is that these `Check`s are almost perfectly
+reproducible, since the sandbox they run in is very controlled. So a `Check`
+that fails or passes on one machine will behave the same on any other machine.
+
+This also means that running these checks on a CI system (for example
+[garnix.io](https://garnix.io)) is very easy and requires no additional
+configuration.
+
+## Executables
+
+(See also the [API docs](https://doc.deno.land/https://garn.io/ts/v0.0.19/mod.ts/~/Executable).)
+
+`Executable`s are commands (usually shell snippets) that are being run on your
+local machine (not in a sandbox). They are based on an underlying `Environment`,
+and can make use of e.g. the tools that `Environment`s provide. You can use
+`Executables` for a variety of purposes:
+
+- Running your projects main executable during development,
+- Running a code formatter,
+- Running a code generator,
+- Spinning up a database for development,
+- etc.
+
+Here's a small example:
+
+```typescript
```

Comment:
There seems to be some `text-shadow` applied to the `<pre>` element on these codeblocks. It makes everything look kind of blurry. Is this applied by `next-mdx-remote`?